Packages/Core/Editor/Generator/AtomGenerator.cs

```diff
@@ -1,86 +1,75 @@
 using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
 using UnityEditor;
 using UnityEngine;
 
 namespace UnityAtoms.Editor
 {
-    [CreateAssetMenu(fileName = "AtomGenerator", menuName = "Unity Atoms/Generation/AtomGenerator", order = 0)]
-    public class AtomGenerator : ScriptableObject
+    [CreateAssetMenu(fileName = nameof(AtomGenerator), menuName = "Atoms/Generators/Atom Generator", order = -10)]
+    public class AtomGenerator : Generator
     {
-        [TextArea] public string FullQualifiedName;
-        public string Namespace;
-        public string BaseType;
-
-        public int GenerationOptions;
-
-        // Referencing Generated Files here:
-        public List<MonoScript> Scripts = new List<MonoScript>(AtomTypes.ALL_ATOM_TYPES.Count);
-
-        private void Reset()
+        private static readonly object populatorsLock = new object();
+        public static event Populator populators
         {
-            for(int i = 0; i < AtomTypes.ALL_ATOM_TYPES.Count; i++)
+            add
             {
-                GenerationOptions |= (1 << i);
+                var generatorGuids = AssetDatabase.FindAssets($"t:{nameof(AtomGenerator)}");
```

Comment:
This part repeats, as well as the loop below it. Extract to private methods or move to base class.

Reply:
Good point, I will make it a private method (the Generator needs to stay clean of this since the AtomGenerator uses this as a kind of "convenience event": no need for the Factories to repeat this FindAssets method every time they want to add their Template.